src/api/auth.ts

```diff
@@ -0,0 +1,25 @@
+import { useMutation } from "@tanstack/react-query";
+import axios from "axios";
+import { ENDPOINTS } from "../constants/apiEndpoints";
+
+export function useSignIn() {
+  return useMutation({
+    mutationFn: (user: { email: string; password: string }) => {
+      return axios.post(ENDPOINTS.SIGN_IN, {
```

Comment:
add types here - I think you can specify request and response types like so: `axios.post<RequestType, ResponseType>` but I might be wrong, please look up docs if that fails